### ISSUE TITLE
[Bug]: Focus indicator gets cut off in CalendarCompat component

### ISSUE THREAD

**Issue by ValentinaKozlova** [microsoft/fluentui#31551]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 12.62 GB / 31.60 GB
  Browsers:
    Edge: Chromium (124.0.2478.105)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://react.fluentui.dev/?path=/docs/compat-components-calendar--default#calendar-month-selection

### Bug Description

**Steps:**
1. Open Month sleection - https://react.fluentui.dev/?path=/docs/compat-components-calendar--default
2. Focus on the grid.

## Actual Behavior
![image](https://github.com/microsoft/fluentui/assets/11574680/c672f8b1-09c2-433a-ac72-495331d11761)

## Expected Behavior
Focus should be fully visible

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by mainframev** [comment]
It's stacking context issue, I guess. When animation is applied, the opacity and transform properties creating a new stacking context. Applying something like `z-index: 1` to Calendar row should fix that, but only on focus, otherwise the row below will overlap it :)

**Comment by ValentinaKozlova** [comment]
Fixed #31596
